core/src/test/java/org/frankframework/mongodb/MongoDbSenderTest.java

```diff
@@ -311,18 +309,39 @@ public List<JsonObject> createStudent(List<String> studentId, String classId, In
 		studentId.forEach(id -> {
 			JsonObjectBuilder builder = Json.createObjectBuilder();
 			builder.add("student_id", id).add("class_id", classId);
-			builder.add("scores", getScores(grades));
+			builder.add("classes", getArrayOfIntegers(grades));
+			builder.add("scores", getScores(grades)); // array of objects
+			builder.add("cities", getCities(grades)); // array of objects with an array
 			students.add(builder.build());
 		});
```

Comment:
The tests in this class only used a quite simple `Student` object. I've now enhanced it with more different types, to cover more real world scenario's.